composer.json

```diff
@@ -25,7 +27,10 @@
         "ext-mbstring": "*",
         "ext-curl": "*",
         "composer/ca-bundle": "~1.0",
-        "symfony/options-resolver": "~2.7|~3.0"
+        "symfony/options-resolver": "~2.7|~3.0",
+        "php-http/httplug": "~1.1",
+        "php-http/curl-client": "~1.7",
+        "zendframework/zend-diactoros": "~1.4"
```

Comment:
Same applies here. You must depend on `psr/http-message-implementation`. This enable users to choose any compatible implementation.